parsers/src/main/java/org/semanticweb/owlapi/rdf/syntax/RDFParser.java

```diff
@@ -84,29 +84,45 @@
 @SuppressWarnings("unused")
 public class RDFParser extends DefaultHandler implements RDFConstants {
 
-    protected static final Locator s_nullDocumentLocator = new LocatorImpl();
-    protected static final SAXParserFactory s_parserFactory = buildFactory();
+    private static ThreadLocal<Locator> locatorThreadLocal = new ThreadLocal<Locator>() {
+        @Override
+        protected Locator initialValue() {
+            return new LocatorImpl();
+        }
+    };
+
+    protected static Locator getLocator() {
+        return locatorThreadLocal.get();
+    }
+    private static ThreadLocal<SAXParserFactory> saxParserFactoryThreadLocal = new ThreadLocal<SAXParserFactory>() {
+
+        @Override
+        protected SAXParserFactory initialValue() {
+            try {
+                System.setProperty("entityExpansionLimit", "100000000");
```

Comment:
There's another thread safety issue somewhere in either the obo parser or
the OWL XML parser ; it shows up once the RDF / XML race is stopped,
manifesting as some warnings that some OWL/XML string can't be parsed by
OBO. However there is no hard failure to cause the unit tests to fail, so
it's more work to catch.

It's most likely an OBO issue.  I will be poking in the OBO stuff later; if
I see something I will fix it, but I won't be looking too hard right now.

The thread fun is trying to make Protégé not do *everything* on the EDT,
which is less than fun when working with GO :)